### ISSUE TITLE
Custom Nodes not loading

### ISSUE THREAD

**Issue by ricardoperucci** [DynamoDS/Dynamo#8232]
## Dynamo version
2.0.0.2757
## Operating system
Windows 10

## What did you do? 

Updated Dynamo version

## What did you expect to see?

Everything working well as it was

## What did you see instead?

Everytime I save any modification inside a Custom Node (or create a new one) and then close Dynamo, when I reopen it, the Custom Node is not loaded. The error message in the console is shown in the first attached image. However, these .dyf files are at the same folder as those that are still working (C:\Users\xxxxxxxx\AppData\Roaming\Dynamo\Dynamo Advance Steel\2.0\definitions).

I’ve realized that, after the update, Custom Nodes (.dyf files) are being saved in a Json format and the older Custom Nodes were all in xml format. It seems that Dynamo is trying to read them in a xml format, but it’s saving them in a Json format.

![screen shot 2017-10-04 at 17 25 36](https://user-images.githubusercontent.com/32527731/31201205-3839b94a-a934-11e7-9d97-8a7249bf8492.png)

![screen shot 2017-10-04 at 17 23 25](https://user-images.githubusercontent.com/32527731/31201211-3c2687e0-a934-11e7-95b7-8b12417e37eb.png)

**Comment by mjkkirschner** [comment]
@ricardoperucci thanks for the report, this is a known issue -  the xml failure is the first attempt before the custom node is loaded in JSON - but the new library UI is not displaying it. It should be fixed soon.

**Comment by mjkkirschner** [comment]
https://github.com/DynamoDS/Dynamo/pull/8233

**Comment by phliberato** [comment]
We are able to customize our Dynamo code to find a temporary workaround to fix it while Dynamo Team don't do it.
We have done some research inside the code and we found some functions which handle Custom Node files.
1 - Save function at WorkspaceViewModel class
2 - ScanNodeHeadersInDirectory function at CustomNodeManager class

Do you have some hint to give us?

**Comment by mjkkirschner** [comment]
@phliberato fix is already in progress in PR I linked above.

**Comment by QilongTang** [comment]
@phliberato Fix is on the way, if urgent, you can cherry-pick that in your local branch

**Comment by ricardoperucci** [comment]
@mjkkirschner  We cherry-picked it, but it is still with problems: It now appears in the nodes' tree, but when I select it for using it on my graph the inputs and outputs are not shown with the node so I can't use it. Maybe is the problem that @phliberato reported at: #8233

**Comment by QilongTang** [comment]
Closing since this should be fixed in the latest built version on master branch, please re-open or file different issue if you find behavior still wrong

**Comment by phliberato** [comment]
@QilongTang I know that you asked to open another file issue, but I think it would be relevant write here the issue that is happening yet.
When I open custom node file by Dynamo open file menu, custom node opens perfectly.(I realized that this action uses `TryGetFunctionWorkspace` function to load  custom node and LibraryServices is not null at this case).
However when I have tried to load custom node using CEFSharp menu or search menu this action fails. At this time the invoked funcion is `TryGetFunctionDefinition` and LibraryServices is null)
